### PR TITLE
Updated Balance model to reflect reality

### PR DESCRIPTION
**Pull request by JanJoris**
After talks with CA, this is what we'll have for now.

### CODE REVIEW

File: definitions/subscriptions/balance.yaml [hunk 1]
```diff
@@ -15,27 +11,16 @@ properties:
       Credit user has to make calls, order extra bundles, ...
     type: number
     format: double
-  billing_rates:
-    $ref: '#/definitions/Rates'
+  out_of_bundle_cost:
```
Comment (bmoelans) [inline definitions/subscriptions/balance.yaml]: since you remove `$ref: '#/definitions/Rates'` is this definition and related yaml still needed?

File: definitions/subscriptions/balance.yaml [hunk 2]
```diff
@@ -2,11 +2,7 @@ type: object
...
 required:
   - product
-  - credit
   - bundles
-  - billing_rates
-  - valid_from
-  - valid_until
 properties:
   product:
     $ref: '#/definitions/ProductLight'
```
Comment (bmoelans) [inline definitions/subscriptions/balance.yaml]: we will get rid of ProductLigth, make product for it

File: definitions/subscriptions/balance.yaml [hunk 3]
```diff
@@ -15,27 +11,16 @@ properties:
...
-    format: date-time
-  valid_until:
-    description: >
-      For prepaid, this value contains the expiration timestamp of the last topup.
-      For postpaid, this is the end of the current billing cycle.
-    type: string
-    format: date-time
+      Only used on postpaid, the current threshold for outside of bundle costs.
+    type: number
+    format: float
```
Comment (bmoelans) [inline definitions/subscriptions/balance.yaml]: for model, what is difference between float and double? I think we return everywhere float unless it is int.

File: definitions/subscriptions/balance.yaml [hunk 4]
```diff
@@ -15,27 +11,16 @@ properties:
...
-      For postpaid, this is the start of the current billing cycle.
-    type: string
-    format: date-time
-  valid_until:
-    description: >
-      For prepaid, this value contains the expiration timestamp of the last topup.
-      For postpaid, this is the end of the current billing cycle.
-    type: string
-    format: date-time
+      Only used on postpaid, the current threshold for outside of bundle costs.
```
Comment (freeduerinckx) [inline definitions/subscriptions/balance.yaml]: So this means it will be 0 for prepaid or the field isn't present?

File: definitions/subscriptions/balance.yaml [hunk 5]
```diff
@@ -15,27 +11,16 @@ properties:
...
     type: number
     format: double
-  billing_rates:
-    $ref: '#/definitions/Rates'
+  out_of_bundle_cost:
     description: >
-      The rates for usage out of bundle. For unlimited items this will contain a rate of 0 Euro.
-      You won't find a bundle of that type within the bundles array.
-  total_out_of_bundle_cost:
+      Only used on postpaid, the costs outside of bundles for the current billing period.
```
Comment (freeduerinckx) [inline definitions/subscriptions/balance.yaml]: So this means it will be 0 for prepaid or the field isn't present?

File: definitions/subscriptions/balance.yaml (hunk 1 above)
Comment (freeduerinckx) [reply definitions/subscriptions/balance.yaml]: I think it is also used in bundle and/or product

Comment (JanJoris) [reply definitions/subscriptions/balance.yaml]: yes, but not for now

File: definitions/subscriptions/balance.yaml (hunk 3 above)
Comment (JanJoris) [reply definitions/subscriptions/balance.yaml]: float, is the one we need to use although there's no real difference for json as there's only float. But for alignment, float is better and clear.

File: definitions/subscriptions/balance.yaml (hunk 5 above)
Comment (JanJoris) [reply definitions/subscriptions/balance.yaml]: We have to decide. 0 is fine actually. Makes this future proof.

File: definitions/subscriptions/balance.yaml (hunk 4 above)
Comment (JanJoris) [reply definitions/subscriptions/balance.yaml]: Always yes.

File: definitions/subscriptions/balance.yaml (hunk 3 above)
Comment (bmoelans) [reply definitions/subscriptions/balance.yaml]: can you than add extra commit where you just replace all double with float to be uniform?

File: assets/sections/unleashed-web-api/templates/resource/balance.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+- **Topup validity**: No workaround, the absence of a bundle will be the definition of a prepaid subscription that needs a topup to benefit from bonus bundles.
+- **Priceplan**: No workaround, but only needed in case you want to add more context towards the prepaid credit you still have. In our own app (Viking App), this is used to generate a VOICE bundle calculated using the national rate for regular traffic. We won't be able to provide this one for now.
+- **Missing bundle types**: Currently we seem to be missing some information on how certain bundles will be returned (roaming, package). We'll update the documentation as soon as we have the information.
+
+## Additions
+
+We do have a live entry for **out of bundle cost** on postpaid including the threshold. Every postpaid account can spend more than it's pre-configured monthly subscription fee, limited to a certain threshold. Per Customer request, this can be adjusted (raised of lowered) and will be shown in the balance response.
```
Comment (bmoelans) [inline assets/sections/unleashed-web-api/templates/resource/balance.md]: of lowered => or lowered

File: assets/sections/unleashed-web-api/templates/resource/balance.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+
+The notion of a **value** bundle is new. When encountering a bundle of this `type`, the `value`, `assigned` and `used` fields will contain EURO content. The **Binge** product for JIM Mobile is configured this way. You can use the value on any other type of choice as long as you've got credit left in this bundle. In the future, we will add the national rates so you could offer a OR view on the other types, indicating to the Customer how much could be used depending on what he or she chooses.
```
Comment (bmoelans) [inline assets/sections/unleashed-web-api/templates/resource/balance.md]: a `OR view`?

Comment (JanJoris) [reply assets/sections/unleashed-web-api/templates/resource/balance.md]: A view containing max SMS or VOICE  or DATA volumes you can use given the balance in the bundle. Adjusted.
